online-shop/src/app/services/auth.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import {Observable, of} from "rxjs";
+import {catchError} from "rxjs/operators";
+import {HttpClient} from "@angular/common/http";
+import {Router} from "@angular/router";
+import {User} from "../model/user";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  loginUrl = '/api/login';
+
+  // store the URL so we can redirect after logging in
+  redirectUrl: string | null = null;
+
+  constructor(
+    private http: HttpClient,
+    private router: Router
+  ) { }
+
+  login(creds: User): Observable<User> {
+    return this.http.post<User>(this.loginUrl, creds)
+      .pipe(
+        catchError(this.handleError('login', creds))
+      );
+  }
+
+  redirect(): void {
+    // Redirect the user
+    this.router.navigate([this.redirectUrl]);
+  }
+
+  private handleError<T>(operation = 'operation', result?: T) {
```

Comment:
You can use this error handling in effects, without console logs. If service fails, dispatch a failure action in effects and handle this in reducer. Because if something is wrong in server side, end customer might be notified. Any non technical person might be aware that we have an error. If you do it like that, you have to inspect page and see the error. This is for your future work, no need to change rn. Maybe handle through a popup component :-?